### PR TITLE
Update to using `wasmparser` 0.55.0

### PR DESCRIPTION
**Pull request by fitzgen**
This includes some SIMD opcode changes and the `AnyRef` to `ExternRef` changes
from the reference types proposal.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1100,7 +1100,7 @@ dependencies = [
  "staticvec",
  "thiserror",
  "typemap",
- "wasmparser",
+ "wasmparser 0.52.2",
```
Comment (alexcrichton) [inline Cargo.lock:1103]: Missing update for lightbeam?

File: Cargo.lock [hunk 2]
```diff
@@ -2057,14 +2057,20 @@ version = "0.52.2"
...
 
+[[package]]
+name = "wasmparser"
+version = "0.55.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "af931e2e1960c53f4a28b063fec4cacd036f35acbec8ff3a4739125b17382a87"
+
 [[package]]
 name = "wasmprinter"
 version = "0.2.4"
```
Comment (alexcrichton) [inline Cargo.lock:2068]: I just published 0.2.5, mind updating that here to try to only use one wasmparser version?

Comment (fitzgen) [reply Cargo.lock:2068]: Sure thing :+1:

**Comment by fitzgen** [comment]
Crap, forgot that the reference types changes mean that we need to update the test suite as well.

**Comment by fitzgen** [comment]
The latest commit temporarily vendors the updated bulk-memory tests that handle `ref.null <type>` introduced in the latest ref types updates. However, we're not going to commit this. Instead, @sunfishcode has volunteered to update the upstream, aggregated spec tests so that they have these changes. Thanks Dan!

**Comment by fitzgen** [comment]
> Instead, @sunfishcode has volunteered to update the upstream, aggregated spec tests so that they have these changes.

This is done now, and the submodule is updated in the latest commit.

**Pull request merged**